pkg/agent/params.go

```diff
@@ -62,6 +62,15 @@ func getParameters(config *datamodel.NodeBootstrappingConfiguration, generatorCo
 		}
 	}
 
+	if properties.CustomConfiguration != nil && properties.CustomConfiguration.KubernetesConfigurations != nil {
+		for componentName, configuration := range properties.CustomConfiguration.KubernetesConfigurations {
```

Comment:
nit: why bother looping over a map when you are going to switch on component name anyway, and hardcode both componentName and customKubeBinaryUrl (i.e., don't use the keys for templating)

Reply:
will use logic like `if val, ok := m["kubelet"]; ok` instead.